lib/internal/url.js

```diff
@@ -243,12 +344,12 @@ class URLSearchParams {
     } else {
       // https://url.spec.whatwg.org/#dom-urlsearchparams-urlsearchparams
       init = toUSVString(init);
-      this[searchParams] = init ? parseParams(init) : [];
+      this.#searchParams = init ? parseParams(init) : [];
     }
   }
 
   [inspect.custom](recurseTimes, ctx) {
-    if (!isURLSearchParams(this))
+    if (typeof this !== 'object' || this === null || !(#searchParams in this))
```

Comment:
Shouldn't this be this != null?

Reply:
No need for `==`, since if `this` was undefined the `typeof` check would have caught it.